src/createSatchel.ts

```diff
@@ -0,0 +1,282 @@
+        __currentMutator: null,
+        // Public functions
+        register: <TAction extends ActionMessage, TReturn = void>(
+            subscriber: Subscriber<TAction, TReturn>
+        ): SubscriberFunction<TAction, TReturn> => {
+            if (getPrivateSubscriberRegistered(subscriber)) {
+                // If the subscriber is already registered, no-op.
+                return subscriber.target;
+            }
+
+            const actionId = getPrivateActionId(subscriber.actionCreator);
+            if (!actionId) {
+                throw new Error(`A ${subscriber.type} can only subscribe to action creators.`);
+            }
+
+            const wrappedTarget =
+                subscriber.type == 'mutator'
+                    ? satchel.__wrapMutatorTarget(subscriber)
+                    : subscriber.target;
+
+            if (!satchel.__subscriptions[actionId]) {
+                satchel.__subscriptions[actionId] = [];
+            }
+
+            satchel.__subscriptions[actionId].push(wrappedTarget);
+            // Mark the subscriber as registered
+            setPrivateSubscriberRegistered(subscriber, true);
+
+            return subscriber.target;
+        },
+        dispatch: (actionMessage: ActionMessage): void => {
+            if (satchel.__currentMutator) {
+                throw new Error(
+                    `Mutator (${satchel.__currentMutator}) may not dispatch action (${actionMessage.type})`
+                );
+            }
+
+            transaction(satchel.__dispatchWithMiddleware.bind(null, actionMessage));
+        },
+        actionCreator: <
+            T extends ActionMessage = {},
+            TActionCreator extends ActionCreator<T> = () => T
+        >(
+            actionType: string,
+            target?: TActionCreator
+        ): TActionCreator => {
+            return satchel.__createActionCreator(actionType, target, false);
+        },
+        action: <T extends ActionMessage = {}, TActionCreator extends ActionCreator<T> = () => T>(
+            actionType: string,
+            target?: TActionCreator
+        ): TActionCreator => {
+            return satchel.__createActionCreator(actionType, target, true);
+        },
+        getRootStore: (): ObservableMap<any> => {
+            return satchel.__rootStore;
+        },
+        hasSubscribers: (actionCreator: ActionCreator<ActionMessage>) => {
+            return !!satchel.__subscriptions[getPrivateActionId(actionCreator)];
+        },
+        createStore: <T>(key: string, initialState: T): (() => T) => {
+            satchel.__createStoreAction(key, initialState);
+            return () => <T>satchel.getRootStore().get(key);
+        },
+        // Private functions
+        __createActionId: (): string => {
+            return (satchel.__nextActionId++).toString();
+        },
+        __finalDispatch:
+            finalDispatch ??
+            ((actionMessage: ActionMessage): void | Promise<void> => {
```

Comment:
Nit: this line is just hard for my brain to parse. I think you can just remove all the typing since TS can infer it knowing that `__finalDispatch` is a `DispatchFunction`.

```
__finalDispatch:
    finalDispatch ??
        actionMessage => { ... };
```